--- SDL_stb.c.orig	2026-07-30 18:42:47
+++ SDL_stb.c	2026-09-14 22:16:19
@@ -65,19 +65,7 @@
 #include "stb_image.h"
 
 ////////////////////////////////////////////////////////////////////////////
-#define MZ_ASSERT(x) SDL_assert(x)
-//#undef memcpy
-//#define memcpy SDL_memcpy
-//#undef memset
-//#define memset SDL_memset
-#if SDL_BYTEORDER == SDL_LIL_ENDIAN
-#define MINIZ_LITTLE_ENDIAN 1
-#else
-#define MINIZ_LITTLE_ENDIAN 0
-#endif
-#define MINIZ_USE_UNALIGNED_LOADS_AND_STORES 0
-#define MINIZ_SDL_NOUNUSED
-#include "miniz.h"
+#include "SDL_miniz_png_ex.h"
 
 #undef memset
 #endif // SDL_HAVE_STB
@@ -483,12 +471,12 @@
     }
 
     size_t size = 0;
-    void *png = tdefl_write_image_to_png_file_in_memory_ex(surface->pixels, surface->w, surface->h, SDL_BYTESPERPIXEL(surface->format), surface->pitch, &size, 6, MZ_FALSE, plte, plte_size, trns, trns_size);
+    void *png = SDL_miniz_write_image_to_png_file_in_memory_ex(surface->pixels, surface->w, surface->h, SDL_BYTESPERPIXEL(surface->format), surface->pitch, &size, 6, MZ_FALSE, plte, plte_size, trns, trns_size);
     if (png) {
         if (SDL_WriteIO(dst, png, size)) {
             retval = true;
         }
-        mz_free(png); /* calls SDL_free() */
+        SDL_free(png);
     } else {
         SDL_SetError("Failed to convert and save image");
     }
